### PR TITLE
[wasm] Improve handling OOM error

### PR DESCRIPTION
**Pull request by mkhamoyan**
By this PR in case of OOM error we will show users message to decrease `EmccMaximumHeapSize`.

Contributes to https://github.com/dotnet/runtime/issues/84638 and https://github.com/dotnet/runtime/issues/95971

### CODE REVIEW

File: src/mono/browser/runtime/loader/run.ts [hunk 1]
```diff
@@ -496,6 +496,13 @@ async function initializeModules (es6Modules: [RuntimeModuleExportsInternal, Nat
 
         return emscriptenModule;
     });
+    result.catch((error) => {
+        if (error.message && error.message.toLowerCase().includes("out of memory")) {
+            mono_log_error("Memory allocation exceeded. Please increase the memory by adjusting EmccMaximumHeapSize.", error);
```
Comment (ilonatommy) [inline src/mono/browser/runtime/loader/run.ts:501]: Should we inform the user what's the current default? (max int)

**Review by ilonatommy — APPROVED** [review]

File: src/mono/browser/runtime/loader/run.ts (hunk 1 above)
Comment (ilonatommy) [reply src/mono/browser/runtime/loader/run.ts:501]: I did not know it, then it should be enough.

**Review by lambdageek — APPROVED** [review]

**Comment by mkhamoyan** [comment]
Failures are not related.

File: src/mono/wasm/features.md [hunk 2]
```diff
@@ -71,7 +71,7 @@ WebSocket support in NodeJS hosts requires the `ws` npm package.
...
 
-To reduce startup time and increase the odds that your application will work on devices with limited memory, you can set an initial size for the memory allocation, based on an estimate of how much memory your application typically uses. To set an initial memory size, include an MSBuild property like `<EmccInitialHeapSize>16777216</EmccInitialHeapSize>`, where you have changed the number of bytes to an appropriate value for your application. This value must be a multiple of 16384.
+To reduce startup time and increase the odds that your application will work on devices with limited memory, you can set an initial size for the memory allocation, based on an estimate of how much memory your application typically uses. To set an initial memory size, include an MSBuild property like `<EmccInitialHeapSize>16777216</EmccInitialHeapSize>`, where you have changed the number of bytes to an appropriate value for your application. This value must be a multiple of 16384. The default value is `2,147,483,648 bytes`, which may be too large and result in the app failing to start, because the browser refuses to grant it.
```
Comment (maraf) [inline src/mono/wasm/features.md:74]: We have mixed `EmccInitialHeapSize` and `EmccMaximumHeapSize`. ~2GB is default value for `EmccMaximumHeapSize`, but the docs is talking about `EmccInitialHeapSize`. Please update it in a follow-up.

Comment (mkhamoyan) [reply src/mono/wasm/features.md:74]: Thanks for noticing, created  https://github.com/dotnet/runtime/pull/105068

**Comment by pavelsavara** [comment]
thanks

**Pull request merged**